GVRf/Framework/framework/src/main/jni/vulkan/vulkanCore.cpp

```diff
@@ -1027,7 +1038,9 @@ void VulkanCore::InitPipelineForRenderData(const GVR_VK_Vertices* m_vertices, Vu
         }
 
         if(image_type & DEPTH_IMAGE && mAttachments[DEPTH_IMAGE]== nullptr){
-            vkImageBase *depthImage = new vkImageBase(VK_IMAGE_VIEW_TYPE_2D, VK_FORMAT_D16_UNORM, mWidth,
+
+            VkFormat depthFormat =  Renderer::getInstance()->useStencilBuffer() ? VK_FORMAT_D24_UNORM_S8_UINT: VK_FORMAT_D16_UNORM;
+            vkImageBase *depthImage = new vkImageBase(VK_IMAGE_VIEW_TYPE_2D, depthFormat, mWidth,
```

Comment:
is there only one format for stencil attachment?

Reply:
There is GL_DEPTH32F_STENCIL8 or VK_FORMAT_D24_UNORM_S8_UINT, which has an 8 bit stencil component.  But the DepthFormat enum in VrAppSettings.java (based on which we determine if stencil testing is to be enabled) has only DEPTH_24_STENCIL_8. Is there a reason why the other format has not been included ?

Reply:
@sushantojal I guess it originates from the Oculus backend; ovrTextureFormat defines only VRAPI_TEXTURE_FORMAT_DEPTH_24_STENCIL_8.